[Component] [message-box] error TS2339: Property 'value' does not exist on type 'MessageBoxData'

<!-- generated by https://issue.element-plus.org/ DO NOT REMOVE -->

Bug Type: **`Component`**

## Environment

- Vue Version: `3.5.27`
- Element Plus Version: `2.13.2`
- Browser / OS: `Microsoft Edge 144.0.3719.115`
- Build Tool: `Vite`

## Reproduction

### Related Component

- `el-message-box`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IEVsTWVzc2FnZUJveCB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuRWxNZXNzYWdlQm94LnByb21wdCgnbWVzc2FnZScsICd0aXRsZScsIHtcbiAgY29uZmlybUJ1dHRvblRleHQ6ICdPSycsXG4gIGNhbmNlbEJ1dHRvblRleHQ6ICdDYW5jZWwnLFxufSkudGhlbigoeyB2YWx1ZSB9KSA9PiB7fSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+PC90ZW1wbGF0ZT5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L1wiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

```vue
<script setup lang="ts">
import { ElMessageBox } from 'element-plus'

ElMessageBox.prompt('message', 'title', {
  confirmButtonText: 'OK',
  cancelButtonText: 'Cancel',
}).then(({ value }) => {})
</script>
```

### What is Expected?

ElMessageBox.prompt的MessageBoxData应该是有value的

### What is actually happening?

error TS2339: Property 'value' does not exist on type 'MessageBoxData'

### Additional comments

2.13.1不会有这个问题，2.13.2才会出现

<!-- 如果你需要插入截图，请在这粘贴。 -->

--- a/packages/components/message-box/__tests__/message-box.test.ts
+++ b/packages/components/message-box/__tests__/message-box.test.ts
@@ -1,4 +1,3 @@
-// @ts-nocheck
 import { markRaw } from 'vue'
 import { mount } from '@vue/test-utils'
 import { afterEach, describe, expect, it, test, vi } from 'vitest'
@@ -7,6 +6,8 @@ import { triggerNativeCompositeClick } from '@element-plus/test-utils/composite-
 import { QuestionFilled as QuestionFilledIcon } from '@element-plus/icons-vue'
 import MessageBox from '../src/messageBox'
 import { ElMessageBox } from '..'
+
+import type { Action } from '..'
 
 const selector = '.el-overlay'
 const QuestionFilled = markRaw(QuestionFilledIcon)
@@ -45,16 +46,16 @@ describe('MessageBox', () => {
         width: '100px',
       },
     })
-    const msgbox: HTMLElement = document.querySelector(selector)
+    const msgbox: HTMLElement = document.querySelector(selector)!
 
     expect(msgbox).toBeDefined()
     await rAF()
     expect(
-      msgbox.querySelector('.el-message-box__title span').textContent
+      msgbox.querySelector('.el-message-box__title span')?.textContent
     ).toEqual('消息')
     expect(
-      msgbox.querySelector('.el-message-box__message').querySelector('p')
-        .textContent
+      msgbox.querySelector('.el-message-box__message')?.querySelector('p')
+        ?.textContent
     ).toEqual('这是一段内容')
     /** custom inline style */
     expect(
@@ -78,12 +79,12 @@ describe('MessageBox', () => {
       message: '这是一段内容',
     })
     await rAF()
-    const icon = document.querySelector('.el-message-box__status')
+    const icon = document.querySelector('.el-message-box__status')!
 
     expect(icon.classList.contains('el-icon')).toBe(true)
 
     const svg = mount(QuestionFilled).find('svg').element
-    expect(icon.querySelector('svg').innerHTML).toBe(svg.innerHTML)
+    expect(icon.querySelector('svg')!.innerHTML).toBe(svg.innerHTML)
   })
 
   test('html string', async () => {
@@ -93,7 +94,7 @@ describe('MessageBox', () => {
       message: '<strong>html string</strong>',
     })
     await rAF()
-    const message = document.querySelector('.el-message-box__message strong')
+    const message = document.querySelector('.el-message-box__message strong')!
     expect(message.textContent).toEqual('html string')
   })
 
@@ -104,7 +105,7 @@ describe('MessageBox', () => {
         title: '消息',
         message: '这是一段内容',
         distinguishCancelAndClose: true,
-        callback: (action) => {
+        callback: (action: Action) => {
           msgAction = action
         },
       })
@@ -127,9 +128,9 @@ describe('MessageBox', () => {
       type: 'warning',
     })
     await rAF()
-    await triggerNativeCompositeClick(document.querySelector(selector))
+    await triggerNativeCompositeClick(document.querySelector(selector)!)
     await rAF()
-    const msgbox: HTMLElement = document.querySelector(selector)
+    const msgbox: HTMLElement = document.querySelector(selector)!
     expect(msgbox.style.display).toEqual('')
     expect(msgbox.querySelector('.el-icon-warning')).toBeDefined()
   })
@@ -142,10 +143,10 @@ describe('MessageBox', () => {
     await rAF()
     const btn = document
       .querySelector(selector)
-      .querySelector('.el-button--primary') as HTMLButtonElement
+      ?.querySelector('.el-button--primary') as HTMLButtonElement
     btn.click()
     await rAF()
-    const msgbox: HTMLElement = document.querySelector(selector)
+    const msgbox: HTMLElement = document.querySelector(selector)!
     expect(msgbox).toBe(null)
   })
 
@@ -157,7 +158,7 @@ describe('MessageBox', () => {
     await rAF()
     const btnElm = document.querySelector(
       '.el-message-box__btns .el-button--primary'
-    )
+    )!
     const haveFocus = btnElm.isSameNode(document.activeElement)
     expect(haveFocus).toBe(false)
   })
@@ -170,10 +171,10 @@ describe('MessageBox', () => {
     })
     await rAF()
     const inputElm = document
-      .querySelector(selector)
-      .querySelector('.el-message-box__input')
+      .querySelector(selector)!
+      .querySelector('.el-message-box__input')!
     const haveFocus = inputElm
-      .querySelector('input')
+      .querySelector('input')!
       .isSameNode(document.activeElement)
     expect(inputElm).toBeDefined()
     expect(haveFocus).toBe(true)
@@ -186,8 +187,8 @@ describe('MessageBox', () => {
     })
     await rAF()
     const textareaElm = document
-      .querySelector(selector)
-      .querySelector('textarea')
+      .querySelector(selector)!
+      .querySelector('textarea')!
     const haveFocus = textareaElm.isSameNode(document.activeElement)
     expect(haveFocus).toBe(true)
   })
@@ -197,7 +198,7 @@ describe('MessageBox', () => {
     MessageBox({
       title: '消息',
       message: '这是一段内容',
-      callback: (action) => {
+      callback: (action: Action) => {
         msgAction = action
       },
     })
@@ -239,7 +240,7 @@ describe('MessageBox', () => {
   test('beforeClose', async () => {
     let msgAction = ''
     MessageBox({
-      callback: (action) => {
+      callback: (action: Action) => {
         msgAction = action
       },
       title: '消息',
@@ -262,7 +263,7 @@ describe('MessageBox', () => {
     test('resolve', async () => {
       let msgAction = ''
       MessageBox.confirm('此操作将永久删除该文件, 是否继续?', '提示').then(
-        (action) => {
+        (action: Action) => {
           msgAction = action
         }
       )
@@ -272,6 +273,29 @@ describe('MessageBox', () => {
       ) as HTMLButtonElement
       btn.click()
       await rAF()
+      expect(msgAction).toEqual('confirm')
+    })
+
+    test('resolve prompt value', async () => {
+      let inputValue = ''
+      let msgAction = ''
+      MessageBox.prompt('请输入内容', '提示').then(({ value, action }) => {
+        inputValue = value
+        msgAction = action
+      })
+      await rAF()
+      const input = document
+        .querySelector(selector)!
+        .querySelector('.el-message-box__input input') as HTMLInputElement
+      input.value = 'test value'
+      input.dispatchEvent(new Event('input'))
+      await rAF()
+      const btn = document.querySelector(
+        '.el-message-box__btns .el-button--primary'
+      ) as HTMLButtonElement
+      btn.click()
+      await rAF()
+      expect(inputValue).toEqual('test value')
       expect(msgAction).toEqual('confirm')
     })
 
@@ -312,7 +336,7 @@ describe('MessageBox', () => {
         title: 'append to test',
         message: 'append to test',
       })
-      const msgbox: HTMLElement = document.querySelector(`body > ${selector}`)
+      const msgbox: HTMLElement = document.querySelector(`body > ${selector}`)!
       expect(msgbox).toBeDefined()
     })
 
@@ -322,7 +346,7 @@ describe('MessageBox', () => {
         message: 'append to test',
         appendTo: '.not-existing-selector',
       })
-      const msgbox: HTMLElement = document.querySelector(`body > ${selector}`)
+      const msgbox: HTMLElement = document.querySelector(`body > ${selector}`)!
       expect(msgbox).toBeDefined()
     })
 
@@ -334,7 +358,7 @@ describe('MessageBox', () => {
         message: 'append to test',
         appendTo: htmlElement,
       })
-      const msgbox: HTMLElement = htmlElement.querySelector(selector)
+      const msgbox: HTMLElement = htmlElement.querySelector(selector)!
       expect(msgbox).toBeDefined()
     })
 
@@ -347,7 +371,7 @@ describe('MessageBox', () => {
         message: 'append to test',
         appendTo: '.custom-html-element',
       })
-      const msgbox: HTMLElement = htmlElement.querySelector(selector)
+      const msgbox: HTMLElement = htmlElement.querySelector(selector)!
       expect(msgbox).toBeDefined()
     })
   })
@@ -412,7 +436,9 @@ describe('MessageBox', () => {
       })
       await rAF()
       const msgbox: HTMLElement = document.querySelector(selector)!
-      const confirmBtn = msgbox.querySelector('.el-button--primary')!
+      const confirmBtn = msgbox.querySelector(
+        '.el-button--primary'
+      )! as HTMLButtonElement
       const error = msgbox.querySelector('.el-message-box__errormsg')!
 
       expect(inputValidator).toHaveBeenCalledTimes(0)

--- a/packages/components/message-box/src/message-box.type.ts
+++ b/packages/components/message-box/src/message-box.type.ts
@@ -9,7 +9,7 @@ type MessageBoxButtonType = (typeof buttonTypes)[number]
 
 export type Action = 'confirm' | 'close' | 'cancel'
 export type MessageBoxType = '' | 'prompt' | 'alert' | 'confirm'
-export type MessageBoxData = MessageBoxInputData | Action
+export type MessageBoxData = MessageBoxInputData & Action
 export interface MessageBoxInputData {
   value: string
   action: Action

